src/Cns.ts

```diff
@@ -255,6 +262,29 @@ export default class Cns extends NamingService {
     return NetworkConfig.networks[this.network].contracts.Registry.address;
   }
 
+  async getDomainFromTokenId(tokenId: string): Promise<string> {
+    const registryAddress = await this.registryAddress('');
```

Comment:
This may break in the future so ideally you need to call `this.registryAddress(tokenId)` and make sure `registryAddress` can accept both: domain name or token id. Up to you and @DeRain to decide if you want to care about it now.

Reply:
@bogdan Agree. Method should receive tokenId and domainName. this.Registry address will make a query to the network to determine Registry address of specified domain or tokenId.